dlib/cuda/cuda_dlib.cu

```diff
@@ -1624,6 +1624,48 @@ namespace dlib
             }
         }
 
+    // ----------------------------------------------------------------------------------------
+
+        __device__ float cuda_log1pexp(float x)
```

Comment:
This implementation was copied from [utilities.h](https://github.com/davisking/dlib/blob/d766f5e82e650fb2b24bbbb8c5d9322a9fcef36c/dlib/dnn/utilities.h#L16-L29), but I'm not sure if the magic numbers should be different for `float` input?

Reply:
Looking at this, I guess the answer is **yes**: https://github.com/JuliaStats/StatsFuns.jl/blob/0beb42294785787ac58fec247cd5d225a99e6d0f/src/basicfuns.jl#L67

Btw, the first Google hit was [this question](https://cs.stackexchange.com/questions/110798/numerically-stable-log1pexp-calculation) :)

Reply:
Fixed.